block/modes.go

```diff
@@ -50,10 +50,8 @@ func (m *Manager) runAsProposer(ctx context.Context, eg *errgroup.Group) error {
 	m.RunMode = RunModeProposer
 	// Subscribe to batch events, to update last submitted height in case batch confirmation was lost. This could happen if the sequencer crash/restarted just after submitting a batch to the settlement and by the time we query the last batch, this batch wasn't accepted yet.
 	go uevent.MustSubscribe(ctx, m.Pubsub, "updateSubmittedHeightLoop", settlement.EventQueryNewSettlementBatchAccepted, m.UpdateLastSubmittedHeight, m.logger)
-
-	// Sequencer must wait till the DA light client is synced. Otherwise it will fail when submitting blocks.
-	// Full-nodes does not need to wait, but if it tries to fetch blocks from DA heights previous to the DA light client height it will fail, and it will retry till it reaches the height.
-	m.DAClient.WaitForSyncing()
+	// Subscribe to P2P received blocks events (used for P2P syncing).
+	go uevent.MustSubscribe(ctx, m.Pubsub, p2pBlocksyncLoop, p2p.EventQueryNewBlockSyncBlock, m.OnReceivedBlock, m.logger)
```

Comment:
why proposer needs p2p sync? syncing from SL is not enough for proposer init flow?

Reply:
why a proposer cannot sync using blocksync? without this, if a node that's been a proposer at some point  needs to resync from genesis, will require to sync from da, while a node that's never been a proposer can. in both cases they just want to request blocks from p2p to get to the sync height. adding this enables requesting blocks using blocksync when out of sync for both.